### PR TITLE
chore: add not unit tests for plugin-data commands

### PR DESCRIPTION
**Pull request by peternhale**
### What does this PR do?
add not unit tests for plugin-data commands

### What issues does this PR fix or reference?
@W-8856572@

### CODE REVIEW

**Review by shetzel — APPROVED** [review]

**Review by amphro — APPROVED** [review]

File: packages/plugin-data/test/commands/force/data/record/dataRecord.nut.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      ).shellOutput.stdout;
+      const match = createRecordResponse.match('Successfully created record: (001.{15})\\.');
+      expect(match).to.have.lengthOf(
+        2,
+        `could not locate success message in results: "${createRecordResponse as string}`
+      );
+      const recordId = match[1] as string;
+
+      // Get a record
+      let getRecordResponse = execCmd(`force:data:record:get --sobjecttype Account --sobjectid ${recordId}`, {
```
[the remark this answers is not in the document]
Comment (peternhale) [reply packages/plugin-data/test/commands/force/data/record/dataRecord.nut.ts:122]: @WillieRuemmele many of these are in unit tests. Is it really necessary to repeat those here?

File: packages/plugin-data/test/commands/force/data/soql/query/dataSoqlQuery.nut.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        "SELECT Id, Name, Phone, Website, NumberOfEmployees, Industry, (SELECT Lastname, Title, Email FROM Contacts) FROM Account  WHERE Name LIKE 'SampleAccount%' limit 1";
+
+      const queryResult = runQuery(query, { ensureExitCode: 0, json: false });
+
+      expect(queryResult).to.match(
+        /ID\s+?NAME\s+?PHONE\s+?WEBSITE\s+?NUMBEROFEMPLOYEES\s+?INDUSTRY\s+?CONTACTS.LASTNAME\s+?CONTACTS.TITLE\s+?CONTACTS.EMAIL/g
+      );
+      expect(queryResult).to.match(/Total number of records retrieved: 1\./g);
+    });
+    it('should return successfully when querying ApexClass column SymbolTable using tooling API', () => {
```
[the remark this answers is not in the document]
Comment (peternhale) [reply packages/plugin-data/test/commands/force/data/soql/query/dataSoqlQuery.nut.ts:142]: @WillieRuemmele covered in unit tests.

File: packages/plugin-data/test/commands/force/data/record/dataRecord.nut.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (peternhale) [reply packages/plugin-data/test/commands/force/data/record/dataRecord.nut.ts:122]: Possibly tooling API, although connection is fetched from base class, which returns either vanilla or tooling connection,  and soql query NUT is exercising tooling API.

[the remark this answers is not in the document]
Comment (WillieRuemmele) [reply packages/plugin-data/test/commands/force/data/record/dataRecord.nut.ts:122]: I think there's confusion amongst the team about how broad we want NUTs to be. I personally think that each flag should be tested in a NUT... 🤷‍♂️

File: packages/plugin-data/test/commands/force/data/bulk/dataBulk.nut.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        const accountIds = records.map((account) => account.Id);
+        const idsFile = path.join(testSession.project?.dir ?? '.', 'data', 'deleteAccounts.csv');
+        try {
+          if (fs.existsSync(idsFile)) {
+            fs.unlinkSync(idsFile);
+          }
+          fs.writeFileSync(idsFile, `Id\n${accountIds.join('\n')}\n`);
+        } catch (error) {
+          assert.fail('', '', 'Could not create csv file of ids to delete');
+        }
```
[the remark this answers is not in the document]
Comment (peternhale) [reply packages/plugin-data/test/commands/force/data/bulk/dataBulk.nut.ts:136]: I'll simplify

**Review by amphro — APPROVED** [review]

**Review by RodEsp — APPROVED** [review]
Compiles and passes linting on Windows perfectly :)
Good job there 👍🏼

**Pull request merged**